src/main/java/graphql/validation/rules/DeferredMustBeOnAllFields.java

```diff
@@ -0,0 +1,92 @@
+package graphql.validation.rules;
+
+import graphql.Directives;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.FragmentDefinition;
+import graphql.language.FragmentSpread;
+import graphql.language.InlineFragment;
+import graphql.language.Selection;
+import graphql.language.SelectionSet;
+import graphql.validation.AbstractRule;
+import graphql.validation.ValidationContext;
+import graphql.validation.ValidationError;
+import graphql.validation.ValidationErrorCollector;
+import graphql.validation.ValidationErrorType;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.groupingBy;
+
+public class DeferredMustBeOnAllFields extends AbstractRule {
+
+    private final Map<List<String>, List<Field>> fieldsByPath = new HashMap<>();
```

Comment:
I always prefer LinkedHashMap as the default impl because order of the keys is kept